src/test/java/com/lafaspot/icap/client/IcapClientTest.java

```diff
@@ -303,4 +304,17 @@ public void testLeaseWithValidSessionCreation() throws URISyntaxException, Timeo
         IcapSession sess = pool.lease(10);
         Assert.assertNotNull(sess);
     }
+
+    @Test(enabled = false)
+    public void testScanFileWithZeroSize() throws IcapException, IOException, InterruptedException, ExecutionException, NoSuchAlgorithmException {
+        final String filename = "emptyFile.txt";
+        byte buf[] = new byte[0];
+
+        URI uri = URI.create("icap://localhost:1344");
+        java.util.concurrent.Future<IcapResult> future = client.scanFile(uri, filename, buf);
+        IcapResult r = future.get();
```

Comment:
assert r is non-null
assert that the byte-array is the same as the one you passed in